libraries/ESP8266WiFi/src/WiFiClientSecureBearSSL.h

```diff
@@ -357,6 +357,21 @@ class WiFiClientSecure : public WiFiClient {
 
     // consume bytes after use (see peekBuffer)
     virtual void peekConsume (size_t consume) override { return _ctx->peekConsume(consume); }
+  
+    void keepAlive(uint16_t idle_sec = TCP_DEFAULT_KEEPALIVE_IDLE_SEC, uint16_t intv_sec = TCP_DEFAULT_KEEPALIVE_INTERVAL_SEC, uint8_t count = TCP_DEFAULT_KEEPALIVE_COUNT)
```

Comment:
Thinking over about this, I would request one change.  Can we make the `WiFiClass` implementation of these as `virtual` and then make these methods `override`?

That way we can guarantee the issue you hit in https://github.com/esp8266/Arduino/issues/8939#issuecomment-1584737239 will be avoided if someone passes a WiFiClientSecure around by reference and it gets consumed as a WiFiClient.

Reply:
Thanks!

Reply:
Yes, I done it.